api2-war/src/main/java/eu/europeana/api2/utils/TurtleRecordWriter.java

```diff
@@ -0,0 +1,263 @@
+package eu.europeana.api2.utils;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+import org.apache.jena.datatypes.RDFDatatype;
+import org.apache.jena.rdf.model.Literal;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.Property;
+import org.apache.jena.rdf.model.RDFNode;
+import org.apache.jena.rdf.model.ResIterator;
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.rdf.model.StmtIterator;
+import org.apache.jena.vocabulary.RDF;
+import org.apache.jena.vocabulary.XSD;
+import static org.apache.commons.lang3.StringUtils.*;
+
+public class TurtleRecordWriter
+{
+    public static final int KB          = 1024;
+    public static       int BUFFER_SIZE = 32 * KB;
+
+    private BufferedWriter     _ps ;
+    private Map<String,String> _decl    = new HashMap();
+
+    public TurtleRecordWriter(OutputStream out) throws IOException
+    {
+        _ps  = new BufferedWriter(new OutputStreamWriter(out, "UTF-8")
+                                , BUFFER_SIZE);
+        _decl.clear();
+    }
+
+    public void write(Model m) throws IOException
+    {
+        writeHeader(m);
+        ResIterator iter = m.listSubjectsWithProperty(RDF.type);
+        try {
+            while ( iter.hasNext() ) {  writeResource(iter.next()); }
+            _ps.flush();
+        }
+        finally {
+            iter.close();
+        }
+    }
+
+    
+    /***************************************************************************
+     * Private Methods
+     **************************************************************************/
+    private void writeResource(Resource r) throws IOException
+    {
+        try {
+            Map<String, Property> props = getProperties(r);
+            int                  len   = calcMaxLength(props.keySet());
+            boolean first = true;
+            _ps.newLine();
+            writeValue(r);
+            for ( Map.Entry<String, Property> entry : props.entrySet() )
+            {
+                if ( first ) { first = false;    }
+                else         { _ps.append(" ;"); }
+                _ps.newLine();
+                writePropertyDecl(entry.getKey(), len);
+                writePropertyValues(r, entry.getValue());
+            }
+            _ps.append(" .");
+            _ps.newLine();
+        }
+        catch (IOException e) { throw e; }
+        catch (Throwable   t) {
+            throw new IOException("Unexpected exception while writting: "
+                    + r.getURI(), t);

```

Comment:
Have changed it to RunTimeException

Reply:
My preference would be not catching it. Rule of thumb is "throw early, catch late", so if do not really have to catch don't do it. But the upside of your approach is that we can log the resource where the error occurred.

Reply:
Yes, I have added the logger to log which resource has thrown error. But if not ideal we can remove it.

Reply:
It's still very weird code. We catch IOException and log it, and we catch RuntimeException and rethrow as IOException..... I propose to not catch anything and let the Record API catch and log the exception instead.

Reply:
I have removed it now. Let Recor Api handle it

Reply:
Thanks for removing the catch of RunTimeException. I think we should remove the catch of IOException as well. In ObjectController we already catch IOException, so if we simply remove it everything works fine. Additionally we can modify the logging in ObjectController so that we also output the recordId whenever an IOException is caught. If there ever is such an exception we can quickly reproduce it with that information.